packages/core/src/debug/debug_node.ts

```diff
@@ -77,21 +77,99 @@ export class DebugNode__PRE_R3__ {
 }
 
 /**
+ *
+ * The `DebugElement` provides crucial insights into the component's DOM representation.
+ *
  * @publicApi
+ *
+ * @see [Component testing scenarios](https://angular.io/guide/testing-components-scenarios)
+ * @see [Basics of testing components](https://angular.io/guide/testing-components-basics)
+ * @see [Testing utility APIs](https://angular.io/guide/testing-utility-apis)
  */
 export interface DebugElement extends DebugNode {
+  /** The element tag name, if it is an element. */
   readonly name: string;
+  /**
+   *  A map of property names to property values for an element.
+   *
+   *  This map includes:
+   *  - Regular property bindings (e.g. `[id]="id"`)
+   *  - Host property bindings (e.g. `host: { '[id]': "id" }`)
+   *  - Interpolated property bindings (e.g. `id="{{ value }}")
+   *
+   *  It does not include:
+   *  - input property bindings (e.g. `[myCustomInput]="value"`)
+   *  - attribute bindings (e.g. `[attr.role]="menu"`)
+   */
   readonly properties: {[key: string]: any};
+  /**
+   *  A map of attribute names to property values for an element.
+   */
   readonly attributes: {[key: string]: string|null};
+  /**
+   * A map containing the class names on the element as keys.
+   *
+   * This map is derived from the `className` property of the native elemnt.
+   *
+   * Note: The values of this object will always be `true`. The class key will not appear in the KV
+   * object if it does not exist on the element.
+   *
+   * @see [Element.className](https://developer.mozilla.org/en-US/docs/Web/API/Element/className)
+   */
   readonly classes: {[key: string]: boolean};
+  /**
+   * The inline styles of the native element.
+   *
+   * Will be `null` if there is no `style` property on the `nativeElement`.
```

Comment:
The `nativeElement` looks like an internal term, so I'd propose replacing it with something else:
```suggestion
   * Will be `null` if there is no `style` property on the underlying DOM element.
```